File: TomUlman/week_01/Friday/mta01/mta.js
//
//                              (   (     (
//                               )   )   )
//                                (     )
//                                  ___
//                                 (   )
//                                 /   \
//                                | (_) |
//                              o_|     |_o
//                             /    _^_    \
//                          _  |   (_O_)   |  _
//                          \_[]\_________/[]_/
//                          []---------------[]
//                           \\|||||||||||||//
//                             \\|||||||||//
//                           ([][]\|||||/[][])
//=============================================================================
//================================= MTA =======================================
//================================ 2016 =======================================
//============================= By Tom Ulman ==================================
//============================= GA 13 Sydney ==================================
//=============================================================================
//=================== Feat: Original ASCII Steam Train ========================
//=============================================================================



  var trainLines = {
   "N" : ["Times Square", "34th", "28th", "23rd", "Union Station", "8th"],
   "L" : ["8th", "6th", "Union Station", "3rd", "1st"],
   "6" : ["Grand Central", "33rd", "28th", "23rd", "Union Station", "Astor Place"]
  };

  var trainSystem = function(startingLine, startingStation, endLine, endStation) {
      // ASSIGNS LINE TO INPUT
    var line1 = trainLines[startingLine];
    var line2 = trainLines[endLine];
      // ASSIGNS STATION INDEX TO INPUT
    var stop1 = line1.indexOf(startingStation);
    var stop2 = line2.indexOf(endStation);
    var union1 = line1.indexOf("uS");
    var union2 = line2.indexOf("uS");
    var newArray1 = null;
    var newArray2 = null;
    var allStops = null;

      // FAILSAFE TO MAKE SURE DATA IS ENTERED CORRECTLY
     if (!((trainLines)[startingLine].includes(startingStation)) ||
     !((trainLines)[endLine].includes(endStation))) {
      alert("You have entered incorrect data, the stations and lines don't match");
     return;
    }

      // TRAVELLING TO AND FROM THE SAME STATION
     if ((line1 === line2) && (stop1 === stop2)) {
       alert("You are hereby fordidden to ride the rail network due to your stupidity");
       allStops = ("0");

      // TRAVELLING ALONG THE SAME LINE
    } else if (line1 === line2) {
       if (stop1 < stop2){
       allStops = (stop2 - stop1);
       newArray1 = line1.slice(stop1, stop2);

    } else if (stop1 > stop2) {
      line1.reverse();
      stop1 = line1.indexOf(startingStation);
      stop2 = line2.indexOf(endStation);
      allStops = (stop2 - stop1);
      newArray2 = line2.slice(stop2, stop1);
    }

    console.log("You are starting from ", startingLine, " Line, ", startingStation, " Station.");
    console.log(" you must travel through ", newArray1, " to reach your final destination at " + endStation);

    } else if (line1 !== line2)  {
      // TRAVELLING FROM THE STARTING LINE
    if (stop1 > union1) {
       line1.reverse();
       stop1 = line1.indexOf(startingStation);
       union1 = line1.indexOf("uS");
       newArray1 = line1.slice(stop1 + 1, union1);
    } else if (stop1 < union1) {
       newArray1 = line1.slice(stop1 + 1, union1);
    }
      // AFTER CHANGING LINES
    if (stop2 < union2) {
       line2.reverse();
       stop2 = line2.indexOf(endStation);
       union2 = line2.indexOf("uS");
       newArray2 = line2.slice(union2 + 1, stop2);
    } else if (stop2 > union2){
       newArray2 = line2.slice(union2 + 1, stop2);
    }

    allStops = ((union1 - stop1) + (stop2 - union2) - 1 );

    console.log("You are starting from ", startingLine, " Line, ", startingStation, " Station.");
    console.log(" you must travel through ", newArray1, " to reach Union Station." );
    console.log("Change at Union Station to ", endLine, " Line.");
    console.log("From Union Station you will travel through ", newArray2, " on the ", endLine, " Line.");
    console.log("You will then arrive at your final destination at ", endStation );

    }

    console.log("The trip will take you " + allStops + " stops in total from your starting station");

};

File: TomUlman/week_04/mta.rb
# The N line has the following stops: Times Square, 34th, 28th, 23rd, Union Square, and 8th
# The L line has the following stops: 8th, 6th, Union Square, 3rd, and 1st
# The 6 line has the following stops: Grand Central, 33rd, 28th, 23rd, Union Square, and Astor Place.
require 'rainbow'
Trainline = {
"N" => ["Times Square", "34th", "28th", "23rd", "Union Square", "8th"],
"L" => ["8th", "6th", "Union Square", "3rd", "1st"],
"6" => ["Grand Central", "33rd", "28th", "23rd","Union Square", "Astor Place"],
}

# Trainline["L"][1] prints 6th
def get_input(l1, s1, l2, s2)

start_station = Trainline[l1].index(s1)
end_station = Trainline[l2].index(s2)

start_line = Trainline[l1]
end_line = Trainline[l2]

start_Us = Trainline[l1].index("Union Square")
end_Us = Trainline[l2].index("Union Square")


#SAME STATION
if l1 == l2 && s1 == s2
puts Rainbow("You are an idiot, go home.").orange
end
#SINGLE LINE TRIP
if l1 == l2 && s1 != s2
travels = Trainline[l1][start_station..end_station]
#SINGLE LINE REVERSE
if start_station > end_station
start_line.reverse!
travels = Trainline[l1][end_station..start_station]

elsif end_station < start_station
#SINGLE LINE FORWARD TRAVEL
travels = Trainline[l1][start_station..end_station]
end
puts "your journey starts on the #{l1} line at #{s1} station"
puts " you will travel through #{travels} to reach"
puts "your destination will then end on the #{l2} line at #{s2} station"
end


#
# MAKE SURE TO LIST THE AMOUNT OF STOPS
#

#TRAVELLING FORWARD ACCROSS LINES
if l1 != l2
travels_one = Trainline[l1][start_station..start_Us]
travels_two = Trainline[l2][end_station..end_Us]

# REVERSE Line1 to Union SQUARE
if start_station > start_Us
start_line.reverse!
travels_one = Trainline[l1][start_Us..start_station]
#TRAVELLING FORWARD ACCROSS LINE2
elsif start_station < start_Us
travels_one = Trainline[l1][start_station..start_Us]
end
# REVERSE Line2 to Union SQUARE
# FIX REVERSE FOR LINE 2
if end_station > end_Us
end_line.reverse!
travels_two = Trainline[l2][end_Us..end_station]
#TRAVELLING FORWARD ACCROSS LINE1
elsif end_station < end_Us
travels_two = Trainline[l2][end_station..end_Us]
end

end
puts "your journey starts on the #{l1} line at #{s1} station"
puts " you will travel through #{travels_one} to reach"
puts "Union Station. You will switch to #{l2}"
puts "From here you will travel through #{travels_two}"
puts "your destination will then end on the #{l2} line at #{s2} station"
end


require 'pry'
binding.pry

end

require 'pry'
binding.pry
